pdfminer/layout.py

```diff
@@ -522,13 +563,97 @@ def __init__(self, bbox):
         self.groups = None
         return
 
-    # group_objects: group text object to textlines.
-    def group_objects(self, laparams, objs):
+    def hsplitted(self, obj0, obj1, splitobjs, cell_margin):
+        ''' detect that between obj0 and obj1 there is a horizontal
+        split line.
+
+        # ARGUMENTS:
+            obj0: a LTTextLineHorizontal, LTTextLineVertical, or LTChar Object
+            obj1: a LTTextLineHorizontal, LTTextLineVertical, or LTChar Object
+            splitobjs: list -- each elememnt is a LTRect Object
+            cell_margin: float -- the distance given for allowing
+                additional matches
+
+        # RETURNS:
+            boolean
+        '''
+        cross = False
+        for r in splitobjs:
+            # line cuts through the overlap of the two objs horizontally
+            if ((r.x0 <= max(obj0.x0, obj1.x0))
+                    and (r.x1 >= min(obj0.x1, obj1.x1))):
```

Comment:
Is there some way you could improve the readability of this logic? Perhaps by adding a methods to `LTComponent`?

Reply:
Sorry for the wait in the response.

This logic is required when parsing tables for which text is not spaced or aligned consistently within a cell.  For example, consider a complex periodic table such as [this](https://upload.wikimedia.org/wikipedia/commons/4/4d/Periodic_table_large.svg). The atomic number is listed in the upper right corner of a cell of the table, the standard atomic weight in the top left. Using the current logic, the layout parser will group  the atomic number of one element and atomic weight of another element (across a dividing line). This new logic will ensure that all the values within a cell are grouped together and respect the boundaries drawn by lines.  Thus, a given text box would contain only the information for one element.

As for speed, I have tried to write the change in a way to not interrupt the original logic.  Thus, cell_margin = None, the layout parser will perform at identical speed.

I will look at adding the method to LTComponent.  However, most of these methods don't look like they will fit the bill.  I will continue looking into it.

Reply:
I have hacked up most of the logic (mathematically it could be vastly simplified).  I didn't want to add it to LTComponent as this method usability to just textboxes and textlines and LTChar.

As a result, it should be much more readable.  What are your thoughts on this new revision?